lib/core_ext/string/normalise_whitespace.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class String
+  # Normalises whitespace in a string by removing leading and trailing whitespace,
+  # replacing multiple spaces with a single space, and returning nil if the result is empty.
```

Comment:
I'd suggest we probably don't need this comment. In general the approach I take is that if a comment like this is required then the code isn't clear enough. In this case, I think the code is pretty clear in what it does so I don't think the comments adds any value.